share/spice/skyscanner_flight_search/skyscanner_flight_search.js

```diff
@@ -0,0 +1,154 @@
+        var build_flight_route = function (current_route, index, response) {
+            var price = "";
+            var price_age = "";
+            var destination_city = "";
+            var destination_city_image = "";
+            var destination_airport = "";
+            var origin_code = "";
+            var destination_code = "";
+            var quote_ref = "";
+            var quote_date_time = "";
+            var outbound_date = "";
+            var return_date = "";
+            var image_array = [];
+
+            var current_route = listOfRoutes[index];
+            var cityId = "SKY:" + placesById[current_route.DestinationId].CityId; 
+            var countryId = "SKY:" + placesById[current_route.DestinationId].SkyscannerCode; 
+            
+            if (response.index[countryId] >=0) {
+                image_array = response.results[response.index[countryId]].images;
+            } 
+            
+            if (Array.isArray(image_array) && image_array.length > 0) {
+                destination_city_image = image_array[0].url;
+            } else {
+                destination_city_image = "http://www.skyscanner.net/images/opengraph.png";
+            }
+
+            price = current_route.Price; 
+            price_age = current_route.QuoteDateTime;
+            // can use CityName or Name here:
+            destination_city = placesById[current_route.DestinationId].Name;
+            destination_airport = placesById[current_route.DestinationId].SkyscannerCode;
+            destination_code = destination_airport;
+            origin_code = placesById[current_route.OriginId].SkyscannerCode;
+            quote_ref = (current_route.QuoteIds) ? ((current_route.QuoteIds.length >1) ? current_route.QuoteIds[0]: current_route.QuoteIds) : '';
+            quote_date_time = (quote_ref != '') ? quotesById[quote_ref].QuoteDateTime : "N/A";
+
+            outbound_date = (quote_ref != '') ? quotesById[quote_ref].OutboundLeg.DepartureDate : "N/A";
+            return_date = (quote_ref != '') ? quotesById[quote_ref].InboundLeg.DepartureDate : "N/A";
+
+            return {
+                flight_price: price,
+                flight_price_age: price_age,
+                flight_destination_city: destination_city,
+                flight_destination_city_image: destination_city_image,
+                flight_destination_airport: destination_airport,
+                flight_destination_code: destination_code,
+                flight_origin_code: origin_code,
+                flight_quote: quote_ref,
+                flight_datetime: quote_date_time,
+                flight_outbound_date: outbound_date,
+                flight_return_date: return_date
+            };
+        }
+        var settings = {
+            "async": true,
+            "crossDomain": true,
+            "url": "https://gateway.skyscanner.net/travel-api/v1/entities?external_ids=" + listOfCountryIds + "&enhancers=images&apikey=09fd8de5844d4b1d982a320ad5dee5b8",
```

Comment:
@samheyman we can't make this API call directly to SkyScanner from the Client as we consider that a Privacy breach (their IP would be exposed)

Fortunately there's an easy fix! You can create an additional endpoint in the Perl, which let's you define another API endpoint to be proxied.

```perl
spice alt_to => {
    sky_scanner_entities => {
        to => 'https://gateway.skyscanner.net/travel-api/v1/entities?external_ids=$1&enhancers=images&apikey={{ENV{DDG_SPICE_SKYSCANNER_APIKEY}}}'
        proxy_cache_valid => '418 1d' #disable api response caching, if needed
    }
}
```

Then from the JS you can do

```js
$.getJSON('/js/spice/sky_scanner_entities/' + listOfCountryIds, function(data) {
    // do work
    DDG.require('moment.js', function() {
        Spice.add({ ... });
    });
});
```